libtock/led.c

```diff
@@ -1,17 +1,37 @@
 #include "led.h"
 
 int led_count(void) {
-  return command(DRIVER_NUM_LEDS, 0, 0, 0);
+  syscall_return_t rval = command2(DRIVER_NUM_LEDS, 0, 0, 0);
+  if (rval.type == TOCK_SYSCALL_SUCCESS_U32) {
+    return rval.data[0];
+  } else {
+    return 0;
+  }
 }
 
 int led_on(int led_num) {
-  return command(DRIVER_NUM_LEDS, 1, led_num, 0);
+  syscall_return_t rval = command2(DRIVER_NUM_LEDS, 1, led_num, 0);
+  if (rval.type == TOCK_SYSCALL_SUCCESS) {
+    return 0;
+  } else {
+    return 1;
+  }
 }
 
 int led_off(int led_num) {
-  return command(DRIVER_NUM_LEDS, 2, led_num, 0);
+  syscall_return_t rval = command2(DRIVER_NUM_LEDS, 2, led_num, 0);
+  if (rval.type == TOCK_SYSCALL_SUCCESS) {
+    return 0;
+  } else {
+    return 1;
+  }
 }
 
 int led_toggle(int led_num) {
-  return command(DRIVER_NUM_LEDS, 3, led_num, 0);
+  syscall_return_t rval = command2(DRIVER_NUM_LEDS, 3, led_num, 0);
+  if (rval.type == TOCK_SYSCALL_SUCCESS) {
+    return 0;
+  } else {
+    return 1;
+  }
```

Comment:
These just return 1 rather than the actual error codes in the case of failure, so are different than the proposed changes in #145 . Not sure which I think is better, just pointing that out

Reply:
Good point. I might have encountered a bug when running a long-term test; I am trying to reproduce it now. Once that resolves I'll fix this.